internal/scheduler/schedulerobjects/podutils.go

```diff
@@ -31,93 +23,197 @@ func (req *PodRequirements) GetAffinityNodeSelector() *v1.NodeSelector {
 	return nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution
 }
 
-// SchedulingKey returns the canonical hash of the scheduling requirements of a job.
+// SchedulingKeyGenerator is used to generate scheduling keys efficiently without requiring allocs.
```

Comment:
Given that this still uses heap-allocated buffers, I have a weak preference for being more precise:

```suggestion
// SchedulingKeyGenerator is used to generate scheduling keys efficiently by sharing serialisation buffers across calls to the Key method.
```